internal/storage/block_storage.go

```diff
@@ -413,10 +396,10 @@ func GetCurrencyKey(currency *types.Currency) string {
 // BalanceChange represents a balance change that affected
 // a *types.AccountIdentifier and a *types.Currency.
 type BalanceChange struct {
-	Account    *types.AccountIdentifier
-	Currency   *types.Currency
-	Block      *types.BlockIdentifier
-	Difference string
+	Account    *types.AccountIdentifier `json:"account_identifier,omitempty"`
+	Currency   *types.Currency          `json:"currency,omitempty"`
+	Block      *types.BlockIdentifier   `json:"block_identifier,omitempty"`
+	Difference string                   `json:"difference,omitempty"`
```

Comment:
Curious why `Difference` is a string type? I'm sure there is a good reason, just wondering.

Reply:
All "amounts" in Rosetta are strings so that any arbitrary-sized, signed integer can be represented (the largest precision requirement I've seen so far is `int256`).

It may make sense to change all amount references to `*big.Int`s in a future pass.

Reply:
Given JSON itself doesn't have bignum support ( https://www.ecma-international.org/ecma-262/5.1/#sec-8.5 ) strings make a lot of sense here.